torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3008,6 +3009,100 @@ struct to_ir {
         return std::make_shared<SimpleValue>(
             emitListComprehension(lc, type_hint));
       }
+      case prim::dict: {
```

Comment:
Optional:

These case branches are getting uncomfortably long, it would be more readable if we break them into helper functions.